Adds the path and http methods section to API style guide (#1435)
opensearch-project/documentation-website: STYLE_API_TEMPLATE.md

  *Example usage*: 
  The response provides a value for `time_field`, such as “timestamp.” 

> Use the `json` format without the `curl` command for API calls, unless the call can only be executed in command line.

Reviewer: ```suggestion
Provide a REST API call example in `json` format. Optionally, also include the `curl` command if the call can only be executed in a command line.
```

Depending on where the documentation appears within a section or subsection, heading levels may be adjusted to fit with other content.

1. Name of API (heading level 2)

> 1. Path and HTTP methods (heading level 3)

Reviewer: ```suggestion
```

"If you use the Security plugin, make sure you have the appropriate permissions."
(To set this point in note-style format, follow the text on the next line with {: .note})

> ### Path and HTTP methods

Reviewer: ```suggestion
```

Reviewer: Need to move paragraph and description to the proper section. This PUT example call goes in the Request fields section with the other PUT/POST API examples that have requests fields.

> If your API call has path parameters, query parameters, and request body fields, it is easier to visualize if you include a section with only the URL of the request without the request fields. The following example shows the URL of the Update a Transform Job API:

Reviewer 2: Most of this comment stems from my lack of experience with APIs. But here goes.
- Does this describe a special case where the API includes all three elements: path parameters, query parameters and request body fields? (e.g., as opposed to an API that only has query parameters)
- "...easier to visualize **the call** ..." ?
- Is this section in lieu of the other 3 sections, or is it just offering a visualization of a completely formed call to help with clarity before describing the 3 elements separately in subsequent sections?

Author: @cwillum:
1. Not a special case per se, but a case where the request is complex enough that it's not easy to see where things go. For simple requests it's normally not necessary, but for more complex ones it helps if you show the endpoint with a path parameter or query parameters so that the user can visualize what we're describing here. So, for example, for transform APIs this section helps see what transform_id is:
`PUT _plugins/_transform/<transform_id>`
2. Yes, easier to see what is a path parameter and a query parameter. Not many people will go to the API style guide to see our terminology, and it may be confusing for users to see the term "query parameter" and "path parameter" when, for example, they are used to calling them "url parameters", which is also correct. I've found that people get confused with our current terminology.
3. Not in lieu by any means. The other sections are mandatory, this one is optional, although I would include it in most cases.
Please feel free to suggest edits so that this is clearer! Thank you.

Reviewer 2: @kolchfa-aws Thanks for the feedback and explanation. Not sure if this makes things muddier. But to follow this through:

"For relatively complex API calls, it's a good idea to provide an example that includes the endpoint and associated parameters or fields so that users can visualize how the request is properly formed. Although optional for basic calls, a realistic example in this section will help illustrate how the endpoint, parameters, and fields fit together in the request and support their description in subsequent sections.

The following example shows the URL of the Update a Transform Job API:"

Just a suggestion. Approving this.

Reviewer: I'm not sure we need this example section above the basic descriptive sections. 
We should have a discussion about this.

> If your API call has path parameters, query parameters, and request body fields, it is easier to visualize if you include a section with only the URL of the request without the request fields. The following example shows the URL of the Update a Transform Job API:

Reviewer: ```suggestion
```

"If you use the Security plugin, make sure you have the appropriate permissions."
(To set this point in note-style format, follow the text on the next line with {: .note})

### Path and HTTP methods

> For relatively complex API calls that include path parameters, it's sometimes a good idea to provide an example so that users can visualize how the request is properly formed. This section is optional and includes examples that illustrate how the endpoint and path parameters fit together in the request. The following is an example of this section for the Nodes stats API:

Reviewer 3: ```suggestion
For relatively complex API calls that include path parameters, it's sometimes a good idea to provide an example so that users can visualize how the request is properly formed. This section is optional and includes examples that illustrate how the endpoint and path parameters fit together in the request. The following is an example of this section for the nodes stats API:
```

Reviewer 3: It is "nodes stats API" on the website.
